tests/test_group_by_key.py

```diff
@@ -0,0 +1,106 @@
+from storey import build_flow, Source, Reduce, GroupByKey
+
+import queue
+
+
+def test_group_by_key():
+    def append_and_return(lst, x):
+        lst.append(x)
+        return lst
+
+    controller = build_flow(
+        [
+            Source(),
+            GroupByKey("value", 4, 100),
+            Reduce([], lambda acc, x: append_and_return(acc, x)),
+        ]
+    ).run()
+
+    values_1 = [i for i in range(3)]
+    values_2 = [i for i in range(3)]
+    values_3 = [i for i in range(3)]
+    for i in range(3):
+        controller.emit({"value": values_1[i], "data": values_1[i]})
```

Comment:
You are sending the same key sequentially.
better have tests that emit the keys not in order (like [key1, key1, key2, key3, key2, key1]) and test scenarios where you reach max events for a certain key and scenarios where you reach timeout for a specific key, etc..

Reply:
No problem